doc/how_to/migrate/from_streamlit.md

```diff
@@ -0,0 +1,970 @@
+bplot = pn.bind(plot, data=data, bins=bins_input)
+
+pn.Column(bins, bplot).servable()
+```
+
+Only the `plot` function is rerun when you change the `bins` slider. You specify that
+by *binding* the `plot` function to the `bins` widget using `pn.bind`.
+
+The app looks like
+
+![Panel Basic Interactivity Example](https://assets.holoviz.org/panel/gifs/panel_interactivity_example.gif)
+
+You might notice that the Panel app updates much quicker and more smoothly than the Streamlit app.
+This is due to fundamental differences in architectures. With Panel you will be able to
+develop very smooth and performant apps.
+
+### Multiple Updates Example
+
+This example will show you how to migrate code that produces a single result
+and updates the UI multiple times during the code execution.
+
+#### Streamlit Multiple Updates Example
+
+```python
+import random
+import time
+
+import streamlit as st
+
+
+def calculation_a():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+
+def calculation_b():
+    time.sleep(3.5)
+    return random.randint(-100, 0)
+
+
+st.write("# Calculation Runner")
+option = st.radio("Which calculation would you like to perform?", ("A", "B"))
+st.write("You chose: ", option)
+if st.button("Press to run calculation"):
+    with st.spinner("Running... Please wait!"):
+        time_start = time.perf_counter()
+        result = calculation_a() if option == "A" else calculation_b()
+        time_end = time.perf_counter()
+    st.write(f"""
+Done!
+
+Result: {result}
+
+The function took {time_end - time_start:1.1f} seconds to complete
+"""
+    )
+else:
+    st.write(f"Calculation {option} did not run yet")
+```
+
+The app looks like
+
+![Streamlit multiple updates example](https://assets.holoviz.org/panel/gifs/streamlit_runner_example.gif)
+
+#### Panel Multiple Updates Example
+
+With Panel you will use a *generator function* to update a component multiple times during code execution.
+
+```python
+import time
+import random
+from textwrap import dedent
+
+import panel as pn
+
+pn.extension(sizing_mode="stretch_width", template="bootstrap")
+pn.state.template.param.update(site="Panel", title="Calculation Runner")
+
+
+def notify_choice(calculation):
+    return f"You chose: {calculation}"
+
+
+def calculation_a():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+
+def calculation_b():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+
+def run_calculation(running, calculation):
+    if not running:
+        yield "Calculation did not run yet"
+        return  # This will break the execution
+    yield pn.Row(
+        pn.indicators.LoadingSpinner(value=True, width=50, height=50, align="center"),
+        "Running... Please wait!",
+        align="center",
+    )
+    if calculation == "A":
+        func = calculation_a
+    else:
+        func = calculation_b
+    time_start = time.perf_counter()
+    result = func()
+    time_end = time.perf_counter()
+    yield dedent(
```

Comment:
Don't understand your use of dedent, it has no effect.

Reply:
I got it from Andrew. I've tested and can see it has no effect and is not needed. Also when I add newlines between the 3 last lines.